src/modules/accommodations/accommodations.service.ts

```diff
@@ -18,8 +28,14 @@ export class AccommodationsService {
     @InjectRepository(Wishlist)
     private readonly wishlistRepository: Repository<Wishlist>,
 
+    @InjectRepository(Amenity)
+    private readonly amenityRepository: Repository<Amenity>,
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
+    @InjectRepository(AccommodationCategory)
+    private readonly categoryRepository: Repository<AccommodationCategory>,
+    private readonly amenitiesService: AmenitiesService,
+    private readonly categoryService: CategoriesService,
```

Comment:
create getCategoryById function in category service and you can use in this service. So you don't need to inject categpryRepository here.